src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1214,6 +1220,107 @@ private IArrayInitializerOperation CreateBoundArrayInitializationOperation(Bound
             return new ArrayInitializerOperation(elementValues, _semanticModel, syntax, isImplicit);
         }
 
+        // PROTOTYPE: Decide public API shape.
+        private IOperation CreateBoundArrayOrSpanCollectionLiteralExpression(BoundArrayOrSpanCollectionLiteralExpression boundCollectionLiteralExpression)
+        {
+            SyntaxNode syntax = boundCollectionLiteralExpression.Syntax;
+            ITypeSymbol? collectionType = boundCollectionLiteralExpression.GetPublicTypeSymbol();
+            bool isImplicit = boundCollectionLiteralExpression.WasCompilerGenerated;
+            var spanConstructor = boundCollectionLiteralExpression.SpanConstructor.GetPublicSymbol();
+            var initializer = new ArrayInitializerOperation(
+                CreateFromArray<BoundExpression, IOperation>(boundCollectionLiteralExpression.Initializers),
+                _semanticModel,
+                syntax,
+                isImplicit: true);
+            ImmutableArray<IOperation> dimensionSizes = ImmutableArray.Create<IOperation>(new LiteralOperation(
+                _semanticModel,
+                syntax,
+                _semanticModel.Compilation.GetSpecialType(SpecialType.System_Int32),
+                ConstantValue.Create(boundCollectionLiteralExpression.Initializers.Length),
+                isImplicit: true));
+            if (spanConstructor is null)
+            {
+                return new ArrayCreationOperation(
+                    dimensionSizes,
+                    initializer,
+                    _semanticModel,
+                    syntax,
+                    collectionType,
+                    isImplicit: isImplicit);
+            }
+            else
+            {
+                var array = new ArrayCreationOperation(
```

Comment:
It feels like this is "tipping our hand" that the span is backed by an array, when the implementation might not always work this way. Is that fine?

Reply:
Added a PROTOTYPE comment.